go/vt/vtgate/vindexes/numeric_static_map.go

```diff
@@ -52,19 +53,52 @@ func init() {
 
 // NewNumericStaticMap creates a NumericStaticMap vindex.
 func NewNumericStaticMap(name string, params map[string]string) (Vindex, error) {
-	jsonPath, ok := params["json_path"]
-	if !ok {
-		return nil, errors.New("NumericStaticMap: Could not find `json_path` param in vschema")
+	jsonStr, jsok := params["json"]
+	jsonPath, jpok := params["json_path"]
+
+	if !jsok && !jpok {
+		return nil, errors.New("NumericStaticMap: Could not find either `json_path` params in vschema")
 	}
 
-	lt, err := loadNumericLookupTable(jsonPath)
-	if err != nil {
-		return nil, err
+	if jsok && jpok {
+		return nil, errors.New("NumericStaticMap: Found both `json` and `json_path` params in vschema")
+	}
+
+	var err error
+	var lt NumericLookupTable
+
+	if jpok {
+		lt, err = loadNumericLookupTable(jsonPath)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if jsok {
+		lt, err = parseNumericLookupTable([]byte(jsonStr))
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	var hashVdx Hashing
+
+	if s, ok := params["fallback_type"]; ok {
+		if err != nil {
```

Comment:
don't we want to check the `err` variable outside of this `if` statement?

Reply:
This check looks completely wrong, not sure what I was thinking here